mypy/semanal.py

```diff
@@ -6776,6 +6784,10 @@ def anal_type(
             prohibit_self_type=prohibit_self_type,
             allow_type_any=allow_type_any,
         )
+        if not a.api.is_stub_file and runtime:
+            a.always_allow_new_syntax = False
+        if runtime is False:
+            a.always_allow_new_syntax = True
```

Comment:
does this account for `__future__.annotations`

Reply:
`always_allow` means stub or future annotations.